aten/src/ATen/native/native_functions.yaml

```diff
@@ -10394,15 +10394,36 @@
   python_module: linalg
   variants: function
 
-- func: linalg_pinv(Tensor self, float rcond=1e-15, bool hermitian=False) -> Tensor
+- func: linalg_pinv.atol_rtol_tensor(Tensor self, *, Tensor? atol=None, Tensor? rtol=None, bool hermitian=False) -> Tensor
+  python_module: linalg
+  variants: function
+
+- func: linalg_pinv.atol_rtol_tensor_out(Tensor self, *, Tensor? atol=None, Tensor? rtol=None, bool hermitian=False, Tensor(a!) out) -> Tensor(a!)
+  python_module: linalg
+  variants: function
+
+# It's not possible to specify both atol and rtol as optional with floats that would work in C++
+# because calling linalg_pinv(self) would be ambiguous, it clashes with the tensor overload.
+# Having one of the float as optional would conflict with the old rcond overload
+# Therefore for floats both values need to be specified and there is no default.
```

Comment:
Make this into a note and cite it in `matrix_rank` as well.

Reply:
I removed this note completely. The problem was fixed with `cpp_no_default_args`.